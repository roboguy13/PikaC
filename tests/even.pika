%generate even

even : Int -> Int;
even n := if (n * 2) == 0 then 1 else 0;
