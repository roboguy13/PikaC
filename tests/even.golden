void even(loc i, loc i1);
void even(loc i, loc i1) {
 WRITE_LOC(i1, 0, (long)i % (long)(2 == 0) ? 1 : 0);
}